### PR TITLE
fetchAll getDocument

### PR DESCRIPTION
**Pull request by fogelito**
Cannot execute queries while other unbuffered queries are active. Consider using PDOStatement::fetchAll(). Alternatively, if your code is only ever going to run against mysql, you may enable query buffering by setting the PDO::MYSQL_ATTR_USE_BUFFERED_QUERY attribute.

### CODE REVIEW

File: src/Database/Adapter/SQL.php [hunk 1]
```diff
@@ -121,12 +121,14 @@ public function getDocument(string $collection, string $id, array $queries = [])
         $stmt->bindValue(':_uid', $id);
         $stmt->execute();
 
-        $document = $stmt->fetch();
+        $document = $stmt->fetchAll();
```
Comment (abnegate) [inline src/Database/Adapter/SQL.php:124]: Is there any other places we need to change to `fetchAll`?

Comment (fogelito) [reply src/Database/Adapter/SQL.php:124]: I added one more place 👍 
For the count (aggregation functions) I do not think we need to do so.

Comment (fogelito) [reply src/Database/Adapter/SQL.php:127]: The error is on line 122 in sentry , not 144 🤔

Comment (abnegate) [reply src/Database/Adapter/SQL.php:127]: > I added one more place 👍 For the count (aggregation functions) I do not think we need to do so.

Let's do it for all `fetch` calls, we can't tell which ones could cause concurrency issues so if we do it for all we can be sure it's not the issue 👌

Comment (fogelito) [reply src/Database/Adapter/SQL.php:127]: Done 👍

File: src/Database/Adapter/MariaDB.php [hunk 2]
```diff
@@ -1181,7 +1183,11 @@ public function count(string $collection, array $queries = [], ?int $max = null,
 
         $stmt->execute();
 
-        $result = $stmt->fetch();
+        $result = $stmt->fetchall();
```
Comment (abnegate) [inline src/Database/Adapter/MariaDB.php:1186]: Let's make sure we use proper casing:

```suggestion
        $result = $stmt->fetchAll();
```

File: src/Database/Adapter/MariaDB.php [hunk 3]
```diff
@@ -1238,7 +1244,11 @@ public function sum(string $collection, string $attribute, array $queries = [],
...
```
Comment (abnegate) [inline src/Database/Adapter/MariaDB.php:1247]: Let's make sure we use proper casing:

```suggestion
        $result = $stmt->fetchAll();
```

File: src/Database/Adapter/SQL.php [hunk 4]
```diff
@@ -81,7 +81,11 @@ public function exists(string $database, ?string $collection): bool
 
         $stmt->execute();
 
-        $document = $stmt->fetch();
+        $document = $stmt->fetchall();
```
Comment (abnegate) [inline src/Database/Adapter/SQL.php:84]: Let's make sure we use proper casing:

```suggestion
        $result = $stmt->fetchAll();
```

File: src/Database/Adapter/SQLite.php [hunk 5]
```diff
@@ -53,7 +53,11 @@ public function exists(string $database, ?string $collection): bool
 
         $stmt->execute();
 
-        $document = $stmt->fetch();
+        $document = $stmt->fetchall();
```
Comment (abnegate) [inline src/Database/Adapter/SQLite.php:56]: Let's make sure we use proper casing:

```suggestion
        $result = $stmt->fetchAll();
```

**Review by abnegate — APPROVED** [review]

**Pull request merged**